gx_suite_definitions/gene_info.ipynb

```diff
@@ -0,0 +1,433 @@
+    "    ExpectColumnValuesToHaveListMembersOfType"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create Expectation Suite for Gene Info Data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Get Example Data File"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "syn = synapseclient.Synapse()\n",
+    "syn.login()\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gene_info_data_file = syn.get(\"syn17015359\").path"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create Validator Object on Data File"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = pd.read_json(gene_info_data_file)\n",
+    "nested_columns = ['target_nominations', 'median_expression', 'druggability', 'ensembl_info']\n",
+    "df = GreatExpectationsRunner.convert_nested_columns_to_json(df, nested_columns)\n",
+    "validator = context.sources.pandas_default.read_dataframe(df)\n",
+    "validator.expectation_suite_name = \"gene_info\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Add Expectations to Validator Object For Each Column"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# ensembl_gene_id\n",
+    "validator.expect_column_values_to_be_of_type(\"ensembl_gene_id\", \"str\")\n",
+    "validator.expect_column_values_to_not_be_null(\"ensembl_gene_id\")\n",
+    "validator.expect_column_value_lengths_to_equal(\"ensembl_gene_id\", 15)\n",
+    "# checks format and allowed chatacters\n",
+    "validator.expect_column_values_to_match_regex(\"ensembl_gene_id\", \"^ENSG\\d{11}$\")\n",
+    "validator.expect_column_values_to_be_unique(\"ensembl_gene_id\")\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# name\n",
+    "validator.expect_column_values_to_be_of_type(\"name\", \"str\")\n",
+    "validator.expect_column_value_lengths_to_be_between(\"name\", 1, 200)\n",
+    "validator.expect_column_values_to_not_be_null(\"name\", mostly=0.70)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# summary\n",
+    "validator.expect_column_values_to_be_of_type(\"summary\", \"str\")\n",
+    "validator.expect_column_values_to_not_be_null(\"summary\", mostly=0.50)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# hgnc_symbol\n",
+    "validator.expect_column_values_to_be_of_type(\"hgnc_symbol\", \"str\")\n",
```

Comment:
Nit: eventually, does it makes sense to have "hgnc symbol" function that adds all of these for data types with "hgnc" validation required?  This way, if you need to update the validation, it updates across all data types?

Reply:
I think one of @JessterB / @jaclynbeck-sage suggested this for either `hgnc_symbol` or `ensembl_gene_id` at one point and I can't remember if there was a reason I didn't want to do it... In this case there is a slight difference to past implementations with the 'mostly` keyword being used in `expect_column_values_to_not_be_null`, but I'm sure we could find a way to account for that if we did break this chunk of code off into one function.

Reply:
I found the thread in Slack that references this:

**Me:** Brad from a purely code-reusability standpoint, does it make sense to implement a generic gx expectation called something like expect_column_is_ensembl_id that just runs all 4 of these expectations without having to type all 4 out in every data set:
```
"validator.expect_column_values_to_be_of_type(\"ensg\", \"str\")\n",
    "validator.expect_column_values_to_not_be_null(\"ensg\")\n",
    "validator.expect_column_value_lengths_to_equal(\"ensg\", 15)\n",
    "# checks format and allowed chatacters\n",
    "validator.expect_column_values_to_match_regex(\"ensg\", \"^ENSG\\d{11}$\")"
```
Since we expect to run this set of 4 expectations on almost every data set? I’m not sure if GX works that way or not though

**Brad:** I think we could do something like that, but it would involve re-implementing the logic for the ensembl_id expectations in a new expectation, so I think for now at least it makes sense to continue doing things this way.
We (in collaboration with the FAIR team) are also getting ready to start work on a Sage-centric great expectations tool and solving this repetition/reusability issue is something we could try to do there.
I also don't necessarily mind a bit of repetition in these cases to be explicit about the expectations we are running for each dataset

----

This was kind of early on in the process, I think we only had expectations for like 2 data sets at that point, so it sounds to me like it was a "we might solve this as part of another project so let's wait".

Reply:
Oh yeah that's right, thanks @jaclynbeck-sage - I didn't love the idea of wrapping multiple expectations up in one, but making a more generic function which calls the expectations would maybe be a little less weird? I also do still prefer being explicit in a lot of ways but if it is going to be repeated this often maybe that becomes less necessary.